include/adm/serial/frame_format_id.hpp

```diff
@@ -0,0 +1,122 @@
+/// @file frame_format_id.hpp
+#pragma once
+
+#include <boost/optional.hpp>
+#include <string>
+#include "adm/detail/named_option_helper.hpp"
+#include "adm/detail/named_type.hpp"
+#include "adm/export.h"
+#include "adm/detail/auto_base.hpp"
+#include "adm/detail/named_type_validators.hpp"
+
+namespace adm {
+
+  /// @brief Tag for FrameFormatIdValue
+  struct FrameIndexTag {};
+  /// @brief NamedType for the frame index portion of a FrameId
+  using FrameIndex = detail::NamedType<unsigned int, FrameIndexTag/*,
+                                       detail::FrameIndexValidator*/>; // TODO fix validators for default constructed NamedTypes
+
+  struct ChunkIndexTag {};
+  /// @brief NamedType for the chunk index portion of a FrameId
+  using ChunkIndex = detail::NamedType<unsigned int, ChunkIndexTag/*,
+                                       detail::ChunkIndexValidator*/>; // TODO fix validators for default constructed NamedTypes
+
+  /// @brief Tag for FrameFormatId
+  struct FrameFormatIdTag {};
+  namespace detail {
+    using FrameFormatIdBase = HasParameters<RequiredParameter<FrameIndex>,
+                                            OptionalParameter<ChunkIndex>>;
+  }  // namespace detail
+
+  /**
+   * @ingroup sadm
+   * @brief Class representation of a frameFormatId element
+   *
+   * \rst
+   * +------------------------+---------------------------------------------------------------+
+   * | Parameter Type         | Description                                                   |
+   * +========================+===============================================================+
+   * | :type:`FrameIndex`     | The index of the current frame within a flow                  |
+   * +------------------------+---------------------------------------------------------------+
+   * | :type:`ChunkIndex`     | The chunk of the current frame if using divided frames        |
+   * +------------------------+---------------------------------------------------------------+
+   * \endrst
+   *
+   * Note that:
+   * - If ChunkIndex is not set, the FF_xxxxxxxx id format will be used
+   * - If ChunkIndex is set, the FF_xxxxxxxx_zz id format will be used
+   * - ChunkIndex only has meaning in divided frame mode
+   */
+  class FrameFormatId : private detail::FrameFormatIdBase,
+                        private detail::AddWrapperMethods<FrameFormatId> {
+   public:
+    typedef FrameFormatIdTag tag;
+
+    using detail::FrameFormatIdBase::get;
+    using detail::FrameFormatIdBase::has;
+    using detail::FrameFormatIdBase::isDefault;
+    using detail::FrameFormatIdBase::set;
+    using detail::FrameFormatIdBase::unset;
```

Comment:
should be private apart from set

Reply:
e90eb85ce094b93d514f5366c969328ee263cb17